pkg/client/rpc.go

```diff
@@ -115,10 +120,24 @@ func newRPC[ResponseType proto.Message](c *RPCClient, i *info.RequestInfo) psrpc
 		defer cancel()
 
 		if i.RequireClaim {
-			serverID, err := selectServer(ctx, claimChan, resChan, o.SelectionOpts)
+			var serverID string
+			for n := 0; n <= o.ClaimRetries; n++ {
+				if n > 0 {
+					if err = c.bus.Publish(ctx, i.GetRPCChannel(), req); err != nil {
+						err = psrpc.NewError(psrpc.Internal, err)
+						return
+					}
+				}
+
+				serverID, err = selectServer(ctx, claimChan, resChan, o.SelectionOpts)
+				if err != nil && !errors.Is(err, psrpc.ErrNoResponse) {
+					return nil, err
+				}
+			}
```

Comment:
if we return an error like `ErrClaimTimeout` when `selectServer` fails the retry interceptor will handle this for us and we can monitor the failure rate using the metrics interceptor. this is already implemented for both unary and streaming rpc.

Reply:
`ErrNoResponse` is only used for claim timeouts, you could still do the above by using `WithClaimRetries(0)` on queue rpcs with a retry interceptor that checks for that error. This way it's the default behavior, and if you want to use retry and metric interceptors it's just an extra line or two

Reply:
why are we duplicating the retry logic here instead of just returning the errors to the interceptor chain to handle? it's not clear to me what benefit we're getting from handling this failure mode differently

Reply:
because queues with nats basically need it, and I don't want adding a retry interceptor to be a requirement for using queue rpcs

Reply:
the library is intended to be transport agnostic. if we need workarounds for specific queue implementations there has to be a better place for them than the core. 

implementors only need to set up clients and servers once but if having a common set of interceptors that we use in cloud is too much boilerplate we can write a generic factory to add the ones we need for standard cloud services...

Reply:
fair enough, I was thinking of it as a request option (ClaimRetries)